Branch-SDK/Branch-SDK/BNCCallbackMap.h

```diff
@@ -0,0 +1,26 @@
+//
+//  BNCCallbackMap.h
+//  Branch
+//
+//  Created by Ernest Cho on 2/25/20.
+//  Copyright © 2020 Branch, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "BNCServerRequest.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BNCCallbackMap : NSObject
+
++ (instancetype)shared;
+
+- (void)storeRequest:(BNCServerRequest *)request withCompletion:(void (^_Nullable)(BOOL success, NSString *statusMessage))completion;
+
+- (BOOL)containsRequest:(BNCServerRequest *)request;
+
+- (void)callCompletionForRequest:(BNCServerRequest *)request withSuccessStatus:(BOOL)status message:(NSString *)statusMessage;
```

Comment:
Did you think about using an `NSError *` instead of `NSString *`? There's lots of stuff on iOS where (in Swift) you often do something like:

```swift
(success: Bool, error: Error?) in

guard success, error == nil else {
    print("Error: \(error!.debugDescription)")
    MessageWindow.show(message: error!.localizedDescription)
    return
}
```

I'm not sure that's called for here, just a thought.

Reply:
Yea you're probably right.  I talked myself out of it for reasons that don't really apply in this situation.  I'll change it in the morning.